tensorly/backend/pytorch_backend.py

```diff
@@ -135,21 +138,29 @@ def solve(matrix1, matrix2):
     return solution
 
 
-def norm(tensor, order):
-    """Computes the l-`order` norm of tensor
+def norm(tensor, order=2, axis=None, **kwds):
+    """Computes the l-`order` norm of tensor.
+
     Parameters
     ----------
     tensor : ndarray
     order : int
+    axis : int
+
     Returns
     -------
-    float
-        l-`order` norm of tensor
+    float or tensor
+        If `axis` is provided returns a tensor.
     """
+    # pytorch does not accept `None` for any keyword arguments
+    if order and order != 'inf':
+        kwds['p'] = order
+    if axis is not None:
+        kwds['dim'] = axis
```

Comment:
I'm thinking for clarity it might be best to explicitly have `p=order` and `dim=axis` in the function calls rather than add them to the dict, especially since we call then call different functions with different parameters. Or at least add the values in the dict only in the relevant part of conditional branching. We could even leave completely the **kwargs out for now, since it is never actually used (if we need other params we can add them then),

Reply:
The reason why I wrote it this way is because of the way Pytorch handles function arguments. (Poorly, in my opinion.) Take a look at the following:

```python
>>> A = torch.arange(9).resize_(3,3)
>>> torch.norm(A, p=1)  # this is fine
36.0
>>> torch.norm(A, p=1, dim=0)  # also fine
  9
 12
 15
[torch.FloatTensor of size 3]
>>> torch.norm(A, dim=0)  # uh oh
TypeError: torch.norm received an invalid combination of arguments - got (torch.FloatTensor, dim=int)
>>> torch.norm(A, p=1, dim=None)  # also wrong. this is the key difference
TypeError: torch.norm received an invalid combination of arguments - got (torch.FloatTensor, dim=NoneType, p=int)
```

So Pytorch essentially doesn't allow keyword arguments. But more key to this issue is that `dim=None` is not allowed.

Reply:
Good point.

Reply:
I will add a comment mentioning something about this